### DISCUSSION TITLE
Upgrade path from 1.x to 2.x

### DISCUSSION THREAD

**Discussion by danFWD** [timber/timber#2769] [Q&A] [answered]
Does anyone have any information on the process from upgrading from Timber 1 to 2 when it’s available? Is it meant to be a seamless transition, will there be minor tweaks necessary to make it work, or are we looking at a full-scale refactor for compatibility?

Thanks in advance.

**Comment by gchtr** [comment, marked as answer]
We have an [Upgrade Guide](https://timber.github.io/docs/v2/upgrade-guides/2.0/) which has details about all the relevant changes made in 2.x.

The transition will not be totally seamless. The 2.x update is actually quite big. How much work is involved depends on how complex and old your Timber theme is. If you don’t use too many advanced features, it could be only be a couple of small changes. It’s definitely not a full-scale refactor. We also added a lot of deprecation and error messages in 2.x that should help you finding the parts in your code that don’t work in 2.x.
